site/content/docs/user/exposing-applications.md

```diff
@@ -0,0 +1,261 @@
+
+```sh
+kubectl expose deployment hello-world --port=80 --target-port=8080 --name=hello-world
+
+service/hello-world exposed
+```
+
+```
+kubectl describe services hello-world
+
+Name:              hello-world
+Namespace:         default
+Labels:            run=load-balancer-example
+Annotations:       <none>
+Selector:          run=load-balancer-example
+Type:              ClusterIP
+IP:                10.106.26.187
+Port:              <unset>  80/TCP
+TargetPort:        8080/TCP
+Endpoints:         10.36.0.1:8080,10.44.0.1:8080
+Session Affinity:  None
+Events:            <none>
+```
+
+### NodePort
+
+> NodePort, as the name implies, opens a specific port on all the Nodes (the VMs), and any traffic that is sent to this port is forwarded to the service.
+
+
+```
+kubectl expose deployment hello-world --type=NodePort --name=hello-world-nodeport
+
+service/hello-world-nodeport exposed
+```
+
+```
+kubectl describe services hello-world-nodeport
+Name:                     hello-world-nodeport
+Namespace:                default
+Labels:                   run=load-balancer-example
+Annotations:              <none>
+Selector:                 run=load-balancer-example
+Type:                     NodePort
+IP:                       10.96.59.230
+Port:                     <unset>  8080/TCP
+TargetPort:               8080/TCP
+NodePort:                 <unset>  30039/TCP
+Endpoints:                10.36.0.1:8080,10.44.0.1:8080
+Session Affinity:         None
+External Traffic Policy:  Cluster
+Events:                   <none>
+```
+
+
+### Ingress
+
+> Ingress is actually NOT a type of service. Instead, it sits in front of multiple services and act as a “smart router” or entrypoint into your cluster.
+
+You can install your ingress controller, i.e. the [NGINX Ingress Controller], and expose it using a NodePort or a Cluster IP. 
+
+### LoadBalancer
+
+> Exposes the service externally using a cloud provider’s or custom load balancer. 
+
+
+## Accesing the application
+
+For some of this methods we'll need the cluster IP addresses
+
+```
+kubectl get nodes -o wide
+
+NAME                 STATUS   ROLES    AGE     VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                                  KERNEL-VERSION     CONTAINER-RUNTIME
+kind-control-plane   Ready    master   7m1s    v1.14.1   172.17.0.4    <none>        Ubuntu Disco Dingo (development branch)   4.9.125-linuxkit   containerd://1.2.6-0ubuntu1
+kind-worker          Ready    <none>   5m58s   v1.14.1   172.17.0.2    <none>        Ubuntu Disco Dingo (development branch)   4.9.125-linuxkit   containerd://1.2.6-0ubuntu1
+kind-worker2         Ready    <none>   5m59s   v1.14.1   172.17.0.3    <none>        Ubuntu Disco Dingo (development branch)   4.9.125-linuxkit   containerd://1.2.6-0ubuntu1
+```
+
+### Kubectl port-forward
+
+This is the most simple one and well documented
```

Comment:
This is the one that I use most of the time maybe worth putting an example, some folks think this is only for pod. But portforward can do a deployment and service also.

For most demos and tutorials I recommend to folks to use a service.
```
kubectl port-forward service/myservice 8080:8080
```
Then to use `locahost:8080` on their browser of if they need a hostname for routing like knative or istio then use a domain lile `xip.io` or `nip.io` using lo `127.0.0.1`
```
curl http://127.1.0.0.1.nip.io
open http://127.1.0.0.1.xip.io
```